tests/lir_tests/convert_lir.rs

```diff
@@ -0,0 +1,481 @@
+    bytes8 %temp.ssa_ir.24 = bytes8(%input) (u)>> bytes8(24);
+    bytes8 %temp.ssa_ir.25 = bytes8 hex"00_00_00_00_00_ff_00_00";
+    bytes8 %temp.ssa_ir.23 = bytes8(%temp.ssa_ir.24) & bytes8(%temp.ssa_ir.25);
+    bytes8 %temp.ssa_ir.3 = bytes8(%temp.ssa_ir.4) | bytes8(%temp.ssa_ir.23);
+    bytes8 %temp.ssa_ir.27 = bytes8(%input) (u)>> bytes8(40);
+    bytes8 %temp.ssa_ir.28 = bytes8 hex"00_00_00_00_00_00_ff_00";
+    bytes8 %temp.ssa_ir.26 = bytes8(%temp.ssa_ir.27) & bytes8(%temp.ssa_ir.28);
+    bytes8 %temp.ssa_ir.2 = bytes8(%temp.ssa_ir.3) | bytes8(%temp.ssa_ir.26);
+    bytes8 %temp.ssa_ir.30 = bytes8(%input) (u)>> bytes8(56);
+    bytes8 %temp.ssa_ir.31 = bytes8 hex"00_00_00_00_00_00_00_ff";
+    bytes8 %temp.ssa_ir.29 = bytes8(%temp.ssa_ir.30) & bytes8(%temp.ssa_ir.31);
+    bytes8 %out = bytes8(%temp.ssa_ir.2) | bytes8(%temp.ssa_ir.29);
+    return bytes8(%out);"#,
+    )
+}
+
+#[test]
+fn test_for_loop() {
+    let cfg_no = 0;
+
+    // read the example.sol file
+    let src = r#"
+		contract test {
+            enum State {
+                Running,
+                Sleeping,
+                Waiting,
+                Stopped,
+                Zombie,
+                StateCount
+            }
+            function get_pid_state(uint64 _pid) pure private returns (State) {
+                uint64 n = 8;
+                for (uint16 i = 1; i < 10; ++i) {
+                    if ((i % 3) == 0) {
+                        n *= _pid / uint64(i);
+                    } else {
+                        n /= 3;
+                    }
+                }
+        
+                return State(n % uint64(State.StateCount));
+            }
+		}"#;
+
+    assert_lir_str_eq(
+        src,
+        cfg_no,
+        r#"private function sol#2 test::test::function::get_pid_state__uint64 (uint64) returns (uint8):
+block#0 entry:
+    uint64 %_pid = uint64(arg#0);
+    uint64 %n = 8;
+    uint16 %i = 1;
+    br block#2;
+
+block#1 body:
+    uint16 %temp.ssa_ir.6 = uint16(%i) (u)% uint16(3);
+    bool %temp.ssa_ir.5 = uint16(%temp.ssa_ir.6) == uint16(0);
+    cbr bool(%temp.ssa_ir.5) block#5 else block#6;
+
+block#2 cond:
+    bool %temp.ssa_ir.7 = uint16(%i) (u)< uint16(10);
+    cbr bool(%temp.ssa_ir.7) block#1 else block#4;
+
+block#3 next:
+    uint16 %temp.4 = uint16(%i) + uint16(1);
+    uint16 %i = uint16(%temp.4);
+    br block#2;
+
+block#4 endfor:
+    uint64 %temp.ssa_ir.9 = uint64(%n) (u)% uint64(5);
+    uint8 %temp.ssa_ir.8 = (trunc uint64(%temp.ssa_ir.9) to uint8);
+    return uint8(%temp.ssa_ir.8);
+
+block#5 then:
+    uint64 %temp.ssa_ir.11 = (zext uint16(%i) to uint64);
+    uint64 %temp.ssa_ir.10 = uint64(%_pid) (u)/ uint64(%temp.ssa_ir.11);
+    uint64 %n = uint64(%n) * uint64(%temp.ssa_ir.10);
+    br block#7;
+
+block#6 else:
+    uint64 %n = uint64(%n) (u)/ uint64(3);
+    br block#7;
+
+block#7 endif:
+    br block#3;"#,
+    )
+}
+
+#[test]
+fn test_test_more_() {
```

Comment:
It's too hard to grasp what this test is supposed to assure. It should either be in the name or deserves explanation in a comment